src/Authentication/Services/SystemUserService.cs

```diff
@@ -54,20 +54,33 @@ public class SystemUserService(
         /// Creates a new SystemUser
         /// The unique Id for the systemuser is handled by the db.
         /// </summary>
-        /// <returns>The SystemUser created</returns>
-        public async Task<SystemUser?> CreateSystemUser(string partyId, SystemUserRequestDto request, int userId)
+        /// <returns>The SystemUser created</returns>    
+        public async Task<Result<SystemUser>> CreateSystemUser(string partyId, SystemUserRequestDto request, int userId)
         {
             RegisteredSystem? regSystem = await _registerRepository.GetRegisteredSystemById(request.SystemId);
             if (regSystem is null)
             {
-                return null;
+                return Problem.SystemIdNotFound;
             }
 
             Party party = await _partiesClient.GetPartyAsync(int.Parse(partyId));
-           
+       
             if (party is null || string.IsNullOrEmpty(party.OrgNumber))
             {
-                return null;
+                return Problem.SystemUserNotFound;
+            }
+
+            ExternalRequestId externalRequestId = new()
+            {
+                OrgNo = party.OrgNumber,
+                SystemId = request.SystemId,
+                ExternalRef = party.OrgNumber // This is the fallback if no ExternalRef is provided, and in L1 this is the same as the OrgNo
```

Comment:
Where are we setting the externalref when its available

Reply:
in a later PR, we need to change the integration from Maskinporten and add a query parameter.